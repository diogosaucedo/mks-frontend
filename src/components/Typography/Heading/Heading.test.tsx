import React from 'react';
import { render, screen } from '@testing-library/react';
import { Heading } from '.';

describe('Heading component', () => {
  it('should render and display properties correctly', () => {
    const text = 'This is a heading';
    const type = 'h2';
    const width = '100%';
    const height = '24px';
    const fontFamily = 'Arial';
    const fontSize = '16px';
    const fontWeight = 600;
    const color = 'black';
    const textAlign = 'center';
    const textDecoration = 'none';
    const textTransform = 'uppercase';
    const letterSpacing = '0.05em';
    const textOverflow = 'ellipsis';
    const overflow = 'hidden';
    const wordWrap = 'break-word';

    render(
      <Heading
        type={type}
        width={width}
        height={height}
        fontFamily={fontFamily}
        fontSize={fontSize}
        fontWeight={fontWeight}
        color={color}
        textAlign={textAlign}
        textDecoration={textDecoration}
        textTransform={textTransform}
        letterSpacing={letterSpacing}
        textOverflow={textOverflow}
        overflow={overflow}
        wordWrap={wordWrap}
      >
        {text}
      </Heading>,
    );

    const headingElement = screen.getByText(text);

    expect(headingElement).toBeInTheDocument();
    expect(headingElement.tagName).toBe('H2');
    expect(headingElement).toHaveStyle(`width: ${width}`);
    expect(headingElement).toHaveStyle(`height: ${height}`);
    expect(headingElement).toHaveStyle(`font-family: ${fontFamily}`);
    expect(headingElement).toHaveStyle(`font-size: ${fontSize}`);
    expect(headingElement).toHaveStyle(`font-weight: ${fontWeight}`);
    expect(headingElement).toHaveStyle(`color: ${color}`);
    expect(headingElement).toHaveStyle(`text-align: ${textAlign}`);
    expect(headingElement).toHaveStyle(`text-decoration: ${textDecoration}`);
    expect(headingElement).toHaveStyle(`text-transform: ${textTransform}`);
    expect(headingElement).toHaveStyle(`letter-spacing: ${letterSpacing}`);
  });

  it('should match snapshot', () => {
    const text = 'This is a heading';
    const type = 'h4';
    const width = '90%';
    const height = '32px';
    const fontFamily = 'Arial';
    const fontSize = '16px';
    const fontWeight = 500;
    const color = 'black';
    const textAlign = 'center';
    const textDecoration = 'none';
    const textTransform = 'uppercase';
    const letterSpacing = '0.05em';
    const textOverflow = 'ellipsis';
    const overflow = 'hidden';
    const wordWrap = 'break-word';

    const { asFragment } = render(
      <Heading
        type={type}
        width={width}
        height={height}
        fontFamily={fontFamily}
        fontSize={fontSize}
        fontWeight={fontWeight}
        color={color}
        textAlign={textAlign}
        textDecoration={textDecoration}
        textTransform={textTransform}
        letterSpacing={letterSpacing}
        textOverflow={textOverflow}
        overflow={overflow}
        wordWrap={wordWrap}
      >
        {text}
      </Heading>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
