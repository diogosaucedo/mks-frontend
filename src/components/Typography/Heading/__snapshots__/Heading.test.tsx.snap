// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component should match snapshot 1`] = `
<DocumentFragment>
  <h4
    class="sc-bcXHqe bAyjRK"
    color="black"
    font-family="Arial"
    font-size="16px"
    font-weight="500"
    height="32px"
    letter-spacing="0.05em"
    overflow="hidden"
    text-decoration="none"
    type="h4"
    width="90%"
  >
    This is a heading
  </h4>
</DocumentFragment>
`;
